src/util/mapbox.js

```diff
@@ -42,8 +42,10 @@ export class RequestManager {
     _skuTokenExpiresAt: number;
     _transformRequestFn: ?RequestTransformFunction;
 
-    constructor(transformRequestFn?: RequestTransformFunction) {
+    constructor(transformRequestFn?: RequestTransformFunction, customAccessToken?: string) {
         this._transformRequestFn = transformRequestFn;
+        if (customAccessToken) config.ACCESS_TOKEN = customAccessToken;
```

Comment:
`config.ACCESS_TOKEN` is used directly to read the `mapboxgl.accessToken` value. Modfiying it here will modify the `mapboxgl.accessToken` and `config.ACCESS_TOKEN` value for all other maps in the same page. The map-specific access token needs to be a member variable of the `RequestManager`

Reply:
>The map-specific access token needs to be a member variable of the RequestManager

This would work well, but turns out that both `RequestManager` and `TelemetryEvent` need this token. The former is invoked by map.js, which is great because it directly handles the custom token from the options object. 

However, the latter is invoked before the map itself gets added, by its child class `TurnstileEvent` here https://github.com/mapbox/mapbox-gl-js/blob/3474c64ba2987d92a2bf38aec24a30a7bb69e155/src/util/mapbox.js#L484. Where is a good place to stow the custom token so that `TelemetryEvent` can refer to it at `event` time?

Reply:
cc @ryanhamley ^ you might have some insight, having worked on turnstile stuff

Reply:
You can store it in the `RequestManager` and then reference it in the `postTurnstileEvent` function since everywhere that `postTurnstileEvent` is called also has access to `RequestManager`. You can see that [I did the same thing with the `postMapLoadEvent`](https://github.com/mapbox/mapbox-gl-js/blob/master/src/source/vector_tile_source.js#L77) which needed the SKU token.